python_tests/pytest_input_generators.py

```diff
@@ -0,0 +1,310 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+# Owner(s): ["module: nvfuser"]
+
+import itertools
+from functools import partial, wraps
+
+import torch
+from torch.testing import make_tensor
+
+from pytest_core import OpInfo, SampleInput, ErrorSample
+from nvfuser import DataType
+
+# TODO Add small value, large value, and extremal-valued samples
+def elementwise_unary_generator(
+    op: OpInfo,
+    dtype: torch.dtype,
+    requires_grad: bool,
+    *,
+    supports_numbers: bool = True,
+    **kwargs,
+):
+    low = None if op.domain.low is None else max(-9, op.domain.low)
+    high = None if op.domain.high is None else min(9, op.domain.high)
+    make_arg = partial(
+        make_tensor,
+        device="cuda",
+        dtype=dtype,
+        low=low,
+        high=high,
+        requires_grad=requires_grad,
+        **kwargs,
+    )
+
+    shapes = (
+        # TODO: restore size zero cases
+        # (0, 2, 1),
+        # (5, 0, 3),
+        (),
+        (11,),
+        (4, 4),
+        (1024, 1024),
+        (64, 64, 64),
+    )
+
+    # Typical inputs
+    for shape in shapes:
+        yield SampleInput(make_arg(shape))
+
+    # Noncontiguous inputs
+    for shape in shapes:
+        yield SampleInput(make_arg(shape, noncontiguous=True))
+
+    # Arbitrarily strided inputs
+    # shape, strides, offset
+    strided_cases = (
+        ((5, 6, 2), (1, 1, 7), 2),
+        ((5, 5, 4), (1, 1, 7), 2),
+        ((5, 5, 2), (4, 5, 7), 3),
+        ((5, 5, 2), (5, 5, 7), 3),
+        ((5, 5, 2), (5, 5, 5), 3),
+        ((9, 5, 2), (0, 1, 7), 3),
+    )
+
+    for shape, strides, offset in strided_cases:
+        a = make_arg(
+            500,
+        ).as_strided(shape, strides, offset)
+        yield SampleInput(a)
+
+
+def _elementwise_unary_torch(op):
+    @wraps(op)
+    def _fn(x):
+        if isinstance(x, torch.Tensor):
+            return op(x)
+        return op(torch.tensor(x)).item()
+
+    return _fn
+
+
+def define_tensor_generator(op, dtype, requires_grad, **kwargs):
+    yield SampleInput(symbolic_sizes=[-1], contiguity=[True])
+
+
+def define_tensor_error_generator(op, dtype, requires_grad, **kwargs):
+    """
```

Comment:
I would add a few more checks:
1. `symbolic_sizes` should check that non-zeros trigger errors
2.  `contiguity` should be checked for values that are not `None`, `True`, or `False`.

Reply:
Or at least add a TODO.  This isn't a blocker for getting in the basic framework.